src/Microsoft.AspNetCore.Blazor.Browser.JS/src/Rendering/BrowserRenderer.ts

```diff
@@ -162,68 +178,41 @@ export class BrowserRenderer {
     this.attachComponentToElement(childComponentId, containerElement);
   }
 
-  insertText(parent: Element, childIndex: number, textFrame: RenderTreeFramePointer) {
+  private insertText(parent: Element, childIndex: number, textFrame: RenderTreeFramePointer) {
     const textContent = renderTreeFrame.textContent(textFrame)!;
     const newDomTextNode = document.createTextNode(textContent);
     insertNodeIntoDOM(newDomTextNode, parent, childIndex);
   }
 
-  applyAttribute(componentId: number, toDomElement: Element, attributeFrame: RenderTreeFramePointer) {
+  private applyAttribute(componentId: number, toDomElement: Element, attributeFrame: RenderTreeFramePointer) {
     const attributeName = renderTreeFrame.attributeName(attributeFrame)!;
     const browserRendererId = this.browserRendererId;
     const eventHandlerId = renderTreeFrame.attributeEventHandlerId(attributeFrame);
 
+    if (eventHandlerId) {
+      const firstTwoChars = attributeName.substring(0, 2);
+      const eventName = attributeName.substring(2);
+      if (firstTwoChars !== 'on' || !eventName) {
```

Comment:
Do you think we should use the same logic in the `Renderer`? We were discussing whether or to 'track'  all delegates regardless of type, it makes sense that we'd use a check like this also since this is the common pattern for DOM events.

Reply:
Yes, that does seem reasonable. Certainly I agree that whatever rules we have should be consistent across compilation, .NET side, and JS side.